### PR TITLE
New option that allow setting the url of the proxy

### PR DESCRIPTION
**Pull request by magne4000**

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ module.exports = function (options, connectionListener) {
 
 	var app = express();
 	var jsonParser = bodyParser.json();
+        var urlRoot = options.urlRoot || '/api/vm/net';
```
Comment (emersion) [inline api.js]: Please indent with ~~spaces~~ tabs :-P

File: browser.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ var debug = util.debuglog('net');
 var proxy = {
 	protocol: (window.location.protocol == 'https:') ? 'wss' : 'ws',
 	hostname: window.location.hostname,
-	port: window.location.port
+	port: window.location.port,
+        url: '/api/vm/net'
```
Comment (emersion) [inline browser.js]: A more accurate name would be `path` - `url` contains `protocol`, `hostname` and `port`.

File: browser.js [hunk 3]
```diff
@@ -26,9 +27,18 @@ function getProxyOrigin() {
 exports.setProxy = function (options) {
 	options = options || {};
 
-	proxy.protocol = options.protocol;
-	proxy.hostname = options.hostname;
-	proxy.port = options.port;
+	if (options.protocol) {
```
Comment (emersion) [inline browser.js:30]: Tabs please! :-)

**Comment by emersion** [comment]
Thanks for your work! Will merge after these few minor fixes!

File: browser.js (hunk 2 above)
Comment (magne4000) [inline browser.js]: I hesitated between the two :smiley:

**Comment by emersion** [comment]
LGTM, thanks!

**Pull request merged**